### PR TITLE
fix: error code on not valid layer version

### PR DESCRIPTION
**Pull request by ronenkapelian**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                      |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Check for version and return appropriate response:
higher version - Ok
same version - Conflict
lower version - Bad Request

### CODE REVIEW

File: src/layers/models/layersManager.ts [hunk 1]
```diff
@@ -108,9 +108,15 @@ export class LayersManager {
     if (requestedLayerVersion > highestExistsLayerVersion) {
       return JobAction.UPDATE;
     }
-    throw new BadRequestError(
-      `layer id: ${resourceId} version: ${version} product type: ${productType} has already the same or higher version (${highestExistsLayerVersion}) in catalog`
-    );
+    if (requestedLayerVersion === highestExistsLayerVersion) {
+      throw new ConflictError(
+        `layer id: ${resourceId} version: ${version} product type: ${productType} has already the same version (${highestExistsLayerVersion}) in catalog`
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:113]: add tests for each condition

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -157,6 +163,7 @@ export class LayersManager {
   private async validateNotExistsInCatalog(resourceId: string, version?: string, productType?: string): Promise<void> {
     const existsInCatalog = await this.catalog.exists(resourceId, version, productType);
     if (existsInCatalog) {
+      console.log('blabla**********************')
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:166]: remove

File: src/layers/models/layersManager.ts [hunk 3]
```diff
@@ -108,9 +108,15 @@ export class LayersManager {
     if (requestedLayerVersion > highestExistsLayerVersion) {
       return JobAction.UPDATE;
     }
-    throw new BadRequestError(
-      `layer id: ${resourceId} version: ${version} product type: ${productType} has already the same or higher version (${highestExistsLayerVersion}) in catalog`
-    );
+    if (requestedLayerVersion === highestExistsLayerVersion) {
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:111]: fix Prettier

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: src/layers/models/layersManager.ts (hunk 3 above)
Comment (ronenkapelian) [reply src/layers/models/layersManager.ts:111]: done

File: src/layers/models/layersManager.ts (hunk 2 above)
Comment (ronenkapelian) [reply src/layers/models/layersManager.ts:166]: done

File: src/layers/models/layersManager.ts (hunk 1 above)
Comment (ronenkapelian) [reply src/layers/models/layersManager.ts:113]: done -> 2 relevant new tests to check the logic

**Review by asafMasa — APPROVED** [review]

**Pull request merged**